From patchwork Fri Dec  7 03:43:18 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Suraj Jitindar Singh <sjitindarsingh@gmail.com>
X-Patchwork-Id: 10717397
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 2A68E15A6
	for <patchwork-kvm@patchwork.kernel.org>;
 Fri,  7 Dec 2018 03:43:43 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 1BE272EDEC
	for <patchwork-kvm@patchwork.kernel.org>;
 Fri,  7 Dec 2018 03:43:43 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 103C02EDF2; Fri,  7 Dec 2018 03:43:43 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-8.0 required=2.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,FREEMAIL_FROM,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI
	autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id AF8D02EDEC
	for <patchwork-kvm@patchwork.kernel.org>;
 Fri,  7 Dec 2018 03:43:42 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1726030AbeLGDnm (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Thu, 6 Dec 2018 22:43:42 -0500
Received: from mail-pg1-f196.google.com ([209.85.215.196]:41021 "EHLO
        mail-pg1-f196.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1726027AbeLGDnl (ORCPT <rfc822;kvm@vger.kernel.org>);
        Thu, 6 Dec 2018 22:43:41 -0500
Received: by mail-pg1-f196.google.com with SMTP id 70so1088143pgh.8;
        Thu, 06 Dec 2018 19:43:41 -0800 (PST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20161025;
        h=from:to:cc:subject:date:message-id:in-reply-to:references;
        bh=WR6xWqNImrfbGP3JXoxK2OHxDdNcXtai1sAf1pCiEFc=;
        b=hWGZw4Uyuh5FMpqroPmzAH+0hVI4hdqqDjYh2bcx60p9p8cRMgag7q8G75Kc5J3Fym
         7oFW82QsyywiBLVgfrm66QcyW08TZu0C5bYNtkSjM9rlFvgRi78Acis7sifFSzVf9MGv
         zXHhMBr6t5/OBAD3UHmeuhP9gjtdD6GySNBveNNdlicbwZ49A4rrAzcRfqiw2DhI+/6r
         Bq9g1VBK5MFYFAmA3+f7pLz4PxfVvSJN7Up0/umNmOfn9yRxDyO+qzpvfLPU2NGOdmby
         CCL3U2DTLoCd3AKUc3E8L3YQsnf8iHwQoii53r2DSLx2bNphTMtxeeL9KuTKsCzJTdBi
         xkCA==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references;
        bh=WR6xWqNImrfbGP3JXoxK2OHxDdNcXtai1sAf1pCiEFc=;
        b=mKmKDQE3KSKgqdHaPIbxBU4D3cOj82XsdOL7VTOrlVYnTJ3PjWzUrFD/92U1+JY1wi
         1VKyvyWsSVCDHS9UWmmD4/YkJM9qGiTgq/WD3SJYUnrwpZvWzUo44ujrOpdts5LXfl6n
         417Z5RpTJpKBbImbqwREP13e1+NRLZIw/MvQlBrd/sTFks5m1HTY7D2Uwqh/yKOTUapQ
         Qm944N7DLC78nscpseMfRHirWv/WDm7I8dXQqnWjC8+ptSOB4+c/PcRV9EXIKHEpkutd
         qwLF6YF0FHQSFGqnLNDWgYdE1qYOU2cyuhBxI5EU92CrsbBFM1cQ4gn76NnuikoyLYpR
         MnNw==
X-Gm-Message-State: AA+aEWZCaLfRmv1WqrWh3Mu62boKLb+dtMpycbXFaoq4YP+ZcRuEuNay
        AsJMFBvGEqn7qk3fJ8aga5vFQkGm
X-Google-Smtp-Source: 
 AFSGD/Xx+jN99b/DLvSvYlSyOuR8j3s4O8hgfsq6p53Dr+g5Cy7JPfi3jHvzpPuZ2o+Mo4Wl7eYzDw==
X-Received: by 2002:aa7:8549:: with SMTP id y9mr530331pfn.83.1544154220879;
        Thu, 06 Dec 2018 19:43:40 -0800 (PST)
Received: from localhost.localdomain ([1.129.152.198])
        by smtp.gmail.com with ESMTPSA id
 e23sm2262646pfh.68.2018.12.06.19.43.37
        (version=TLS1_2 cipher=ECDHE-RSA-CHACHA20-POLY1305 bits=256/256);
        Thu, 06 Dec 2018 19:43:40 -0800 (PST)
From: Suraj Jitindar Singh <sjitindarsingh@gmail.com>
To: kvm-ppc@vger.kernel.org
Cc: sjitindarsingh@gmail.com, kvm@vger.kernel.org, paulus@ozlabs.org,
        linuxppc-dev@lists.ozlabs.org, aik@ozlabs.ru
Subject: [PATCH] KVM: PPC: Book3S PR: Set hflag to indicate that POWER9
 supports 1T segments
Date: Fri,  7 Dec 2018 14:43:18 +1100
Message-Id: <20181207034326.3564-2-sjitindarsingh@gmail.com>
X-Mailer: git-send-email 2.13.6
In-Reply-To: <20181207034326.3564-1-sjitindarsingh@gmail.com>
References: <20181207034326.3564-1-sjitindarsingh@gmail.com>
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

When booting a kvm-pr guest on a POWER9 machine the following message is
observed:
"qemu-system-ppc64: KVM does not support 1TiB segments which guest expects"

This is because the guest is expecting to be able to use 1T segments
however we don't indicate support for it. This is because we don't set
the BOOK3S_HFLAG_MULTI_PGSIZE flag in the hflags in kvmppc_set_pvr_pr()
on POWER9.

POWER9 does indeed have support for 1T segments, so add a case for
POWER9 to the switch statement to ensure it is set.

Signed-off-by: Suraj Jitindar Singh <sjitindarsingh@gmail.com>
---
 arch/powerpc/kvm/book3s_pr.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/powerpc/kvm/book3s_pr.c b/arch/powerpc/kvm/book3s_pr.c
index 4efd65d9e828..82840160c606 100644
--- a/arch/powerpc/kvm/book3s_pr.c
+++ b/arch/powerpc/kvm/book3s_pr.c
@@ -587,6 +587,7 @@ void kvmppc_set_pvr_pr(struct kvm_vcpu *vcpu, u32 pvr)
 	case PVR_POWER8:
 	case PVR_POWER8E:
 	case PVR_POWER8NVL:
+	case PVR_POWER9:
 		vcpu->arch.hflags |= BOOK3S_HFLAG_MULTI_PGSIZE |
 			BOOK3S_HFLAG_NEW_TLBIE;
 		break;
